packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesController.kt

```diff
@@ -54,12 +54,17 @@ class UpdatesController private constructor(
 
   // TODO: move away from DatabaseHolder pattern to Handler thread
   private val databaseHandlerThread = HandlerThread("expo-updates-database")
+  private var isDatabaseHandlerThreadStarted = false
   private lateinit var databaseHandler: Handler
   private fun initializeDatabaseHandler() {
-    databaseHandlerThread.start()
+    if (!isDatabaseHandlerThreadStarted) {
+      isDatabaseHandlerThreadStarted = true
+      databaseHandlerThread.start()
+    }
     databaseHandler = Handler(databaseHandlerThread.looper)
```

Comment:
Probably want to move this inside the check as well. While I think things will still work since two handlers would share the same looper, it's trickier to think about. In fact, perhaps this method should be written:

```kt
if (databaseHandler == null) {
  databaseHandlerThread.start()
  databaseHandler = Handler(databaseHandlerThread.looper)
}
```

Reply:
Yep, this makes sense and is a little clearer.